CDH2.py

```diff
@@ -0,0 +1,348 @@
+
+        :param self: DNA or RNA sequence
+        '''
+        
+        if (set(self.seq).issubset(self.dna_alphabet) and isinstance(self, DNASequence)) or (set(self.seq).issubset(self.rna_alphabet) and isinstance(self, RNASequence)):
+            return True
+        raise TypeError(f'{self.seq} is not correct nucleic acid')          
+
+
+    def complement(self):
+        """
+        Function return complement sequence.
+        
+        :param self: DNA or RNA sequence
+        :rtype: str
+        :return: complement sequence   
+        """
+        if self.is_alphabet_correct():
+            complement_seq = str()
+            length = len(self.seq)
+            for i in range (length):
+                if self.seq[i] in self.complement_dict:
+                    complement_seq += (self.complement_dict[self[i]])
+            if isinstance(self, DNASequence):
+                return DNASequence(complement_seq)
+            if isinstance(self, RNASequence):
+                return RNASequence(complement_seq)
+        
+    
+    def gc_calculate(self) -> float:
+        """
+        Function return sequence GC-content in percent.
+        
+        :param seq: DNA or RNA sequence
+        :type seq: str
+        :rtype: float
+        :return: GC-contentn percent 
+        """
+        length = len(self.seq)
+        gc_content = 0.0
+        seq_up = self.seq.upper()
+        c = seq_up.count("C")
+        g = seq_up.count("G")
+        gc_content = round(((c+g)/length*100),2)
+        return gc_content
+
+
+class DNASequence(NucleicAcidSequnce):
+    '''
+    Class for DNA sequence
+    '''
+
+    def __init__(self, seq) -> None:
+        super().__init__(seq)
+
+    def transcribe(self):
+        '''
+        Method return transcribed sequence.
+        '''
+
+        if super().is_alphabet_correct():
+            return RNASequence(self.seq.replace('T', 'U').replace('t', 'u'))
+    
+
+class RNASequence(NucleicAcidSequnce):
+    '''
+    Class for RNA sequence
+    '''
+    
+    def __init__(self, seq) -> None:
+        super().__init__(seq)
+
+
+class AminoAcidSequence (BiologicalSequence):
+    '''
+    Class for protein sequence
+    '''
+    
+    def __init__(self, seq):
+        self.seq = seq
+        self.protein_alphabet = set('ACDEFGHIKLMNPQRSTVWY')
+
+
+    def __len__(self):
+        return len(self.seq)
+
+
+    def __getitem__(self, item):
+        return self.seq[item]
+
+
+    def __str__(self):
+        return self.seq
+
+
+    def is_alphabet_correct(self):
+        if set(self.seq).issubset(self.protein_alphabet):
+            return True
+        raise TypeError(f'{self.seq} is not a protein')    
+
+    def calculate_protein_mass(self):
+        '''
+        Method return mass of residues in seq in Da.
+        '''
+        
+        if self.is_alphabet_correct:
+            weights = {'A': 89.09, 'R': 174.20, 'N': 132.12, 'D': 133.10, 'C': 121.15,
```

Comment:
Такая переменная хорошо бы смотрелась в роли классового атрибута вне функции
```suggestion
            AMINOACID_WEIGHTS = {'A': 89.09, 'R': 174.20, 'N': 132.12, 'D': 133.10, 'C': 121.15,
```

Reply:
Согласен
Только тут это теперь не константа. Константы это переменные которые объявляются в самом начале скрипта. Это просто классовый атрибут, поэтому его не надо делать капсом.